cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -329,22 +329,23 @@ public void getAdapter(HttpRequest request, HttpResponder responder,
   }
 
   /**
-   * Starts/stops an adapter
+   * Suspends/resumes an adapter
    */
   @POST
   @Path("/adapters/{adapterId}/{action}")
-  public void startStopAdapter(HttpRequest request, HttpResponder responder,
-                               @PathParam("namespace-id") String namespaceId,
-                               @PathParam("adapterId") String adapterId,
-                               @PathParam("action") String action) {
+  public void suspendResumeAdapter(HttpRequest request, HttpResponder responder,
+                                   @PathParam("namespace-id") String namespaceId,
+                                   @PathParam("adapterId") String adapterId,
+                                   @PathParam("action") String action) {
     try {
-      if ("start".equals(action)) {
-        adapterService.startAdapter(namespaceId, adapterId);
-      } else if ("stop".equals(action)) {
-        adapterService.stopAdapter(namespaceId, adapterId);
+      if ("resume".equals(action)) {
```

Comment:
It should not be suspend/resume. It should be start/stop. We shouldn't expose what we do internally in this high level API.